tilelang/utils/pass_diff_hook.py

```
@@ -0,0 +1,198 @@
+        timestamp = time.strftime("%Y%m%d_%H%M%S")
+        _html_path = os.path.join(diff_dir, f"pass_diff_{timestamp}.html")
+
+    import atexit
+
+    def _flush_html():
+        if _html_path and _html_steps:
+            generate_html = _diff_utils["generate_html"]
+            try:
+                generate_html(_html_steps, _html_path)
+                print(f"\n[pass_diff] HTML report: {_html_path}")
+            except Exception as e:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)